### PR TITLE
Add an implementation of encrypt_file and decrypt_file.

### PR DESCRIPTION
**Pull request by jochumdev**
This allows chunked encoding and decoding of files.

### CODE REVIEW

File: CHANGES.txt [hunk 1]
```diff
@@ -5,7 +5,9 @@ CHANGES
 3.0.1 (unreleased)
 ------------------
 
-- Nothing changed yet.
+- Add an implementation of encrypt_file and decrypt_file.
+  This allows chuncked encoding and decoding of files.
```
Comment (strichter) [inline CHANGES.txt]: typo: chunked

File: src/keas/kmi/facility.py [hunk 2]
```diff
@@ -88,10 +146,44 @@ def decrypt(self, key, data):
             key=encryptionKey, mode=self.CipherMode,
             IV=self.initializationVector)
         # 3. Decrypt the data.
-        text = cipher.decrypt(data)
+        try:
+            text = cipher.decrypt(data)
+        except ValueError:
```
Comment (strichter) [inline src/keas/kmi/facility.py]: This changes the API significantly. Why are you starting to catch ValueError?

File: src/keas/kmi/facility.py [hunk 3]
```diff
@@ -88,10 +146,44 @@ def decrypt(self, key, data):
...
+
         # 4. Remove padding and return result.
         return self._pkcs7Decode(text)
 
+    def decrypt_file(self, key, fsrc, fdst, chunksize=24*1024):
+        """ Decrypts a file using with the given key.
+        Parameters are similar to encrypt_file.
+        """
+        header = str(fsrc.read(2))
+        if header != '.e':
```
Comment (strichter) [inline src/keas/kmi/facility.py]: What about files that start legitimately with ".e"? I do not think you later protect against that.

**Comment by strichter** [comment]
Other than my few comments, this looks pretty good. No tests though?

**Comment by jochumdev** [comment]
@frisi fyi

File: src/keas/kmi/facility.py (hunk 2 above)
Comment (jochumdev) [inline src/keas/kmi/facility.py]: Will remove that try/except block.

File: src/keas/kmi/facility.py (hunk 3 above)
Comment (jochumdev) [inline src/keas/kmi/facility.py]: After a discussion with @frisi we found out that the ".e" marker isn't needed in keas.kmi.

".e" is from https://github.com/zopefoundation/cipher.encryptingstorage and keas.kmi should just fail hard when it gets an unencrypted file on decrypt_file. 

So i'll remove the ".e" marker from encrypt_file and decrypt_file.

Thank you @strichter for pointing this out.

File: CHANGES.txt (hunk 1 above)
Comment (jochumdev) [inline CHANGES.txt]: Thank you!

**Comment by pcdummy** [comment]
I've moved the ".e" thing to cipher.encryptingstorage (where we use keas.kmi).

@strichter can you help me with the tests, idk where to start.

**Comment by jochumdev** [comment]
https://github.com/zopefoundation/cipher.encryptingstorage/pull/1 <-- this is where we use it.

**Comment by jochumdev** [comment]
@strichter Test is done, can you make a release?

**Comment by strichter** [comment]
@pcdummy Do you have a PyPI id that II can make you a maintainer? I assume you know how to release. :-)

**Comment by jochumdev** [comment]
Mine is "pcdummy" and the one of my teammate is "frisi", please add us both.

**Comment by strichter** [comment]
I ended up cuting the release, but I will still add you guys.

**Pull request merged**